nengo/spa/vocab.py

```diff
@@ -379,6 +400,20 @@ def prob_cleanup(self, similarity, vocab_size, steps=10000):
         pcorrect = (1 - perror1) ** vocab_size
         return pcorrect
 
+    def extend(self, keys):
+        """Extends the vocabulary with additional keys.
+
+        Creates and adds the semantic pointers listed in keys to the
+        vocabulary.
+
+        Parameters
+        ----------
+        keys : list of strings
+            List of semantic pointer names to be added to the vocabulary.
+
+        """
+        self.parse('+'.join(keys))
```

Comment:
for clarity, it might be nicer to implement this as

```python
for k in keys:
    self[k]
```

or perhaps even

```python
for k in keys:
    if k not in self.keys:
        self.add(k, self.create_pointer())
```

or even

```python
def extend(self, keys, unitary=False):
    for k in keys:
        if k not in self.keys:
            self.add(k, self.create_pointer(unitary=unitary))
```